test/Templates.Test/Helpers/SeleniumServerFixture.cs

```diff
@@ -0,0 +1,74 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Runtime.InteropServices;
+
+namespace Templates.Test.Helpers
+{
+    public class SeleniumServerFixture : IDisposable
+    {
+        private string _workingDirectory;
+        private Process _serverProcess;
+        private static readonly object _serverLock = new object();
+
+        public SeleniumServerFixture()
+        {
+            _workingDirectory = Directory.GetCurrentDirectory();
+            StartSeleniumStandaloneServer();
+        }
+
+        public void Dispose()
+        {
+            _serverProcess.Kill();
+            _serverProcess.Dispose();
+
+            // Find the java process running selenium-standalone
```

Comment:
I don't quite understand this bit. It looks like on line 26 the `selenium-standalone` process was already killed. Is the issue that we also need to kill the process tree underneath it?

The logic here is probably fine for our use cases but it looks slightly dangerous in that it kills *all* the `java` child processes running on the machine, even if they are unrelated to the tests. Might it be enough to only kill the child processes of `_serverProcess`?

Reply:
Yes, the issue is that line 26 kills the process that started the selenium server but not the underlying java process.

Agreed. On the CI at least, there shouldn't be random java processes running but this is not ideal for sure. The code to get child processes is a little round about but there is a way. Let me try it out and update the PR!